modules/shared_data/src/main/scala/org/elpaca_metagraph/shared_data/types/ConstellationBlockExplorer.scala

```diff
@@ -14,12 +16,9 @@ object ConstellationBlockExplorer {
     amount     : Long,
     timestamp  : String
   ) {
-    override def equals(obj: Any): Boolean = obj match {
-      case that: BlockExplorerTransaction => this.hash == that.hash
-      case _ => false
+    implicit val eqInstance: Eq[BlockExplorerTransaction] = Eq.instance { (a, b) =>
+      a.hash === b.hash
```

Comment:
FYI - when you make such an `Eq` global (as you did) - classes with different properties but same hashes would be treated as equally the same. `List` -> `Set` would deduplicate transactions based on its hashes, so duplicates would be removed and a single transaction will remain. This is generally a good idea but not in all places - hence, it's better to use a locally defined `Eq` instead of global one. That being said, as you defined a global `Eq` for `ConstellationBlockExplorer`, you won't be able to detect that you have two conflicting transactions for example (same hash but different properties).

Reply:
That's what we want for this scenario, avoid the duplications to all occurrences in this class